apps/condo/domains/miniapp/schema/B2BAppAccessRightSet.test.js

```diff
@@ -0,0 +1,650 @@
+                expect(!!deletedAccessRightSet.deletedAt).toBeTruthy()
+            })
+
+            test('Can not hard-delete', async () => {
+                const [app] = await createTestB2BApp(admin)
+                const [createdAccessRightSet] = await createTestB2BAppAccessRightSet(admin, app)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await B2BAppAccessRightSet.delete(admin, createdAccessRightSet.id)
+                })
+            })
+        })
+
+        describe('Support', () => {
+            test('Can create', async () => {
+                const [app] = await createTestB2BApp(admin)
+                const [createdAccessRightSet] = await createTestB2BAppAccessRightSet(support, app)
+
+                expect(createdAccessRightSet).toBeDefined()
+                expect(createdAccessRightSet.id).toMatch(UUID_RE)
+                expect(createdAccessRightSet).toHaveProperty('app.id', app.id)
+            })
+
+            test('Can read everything', async () => {
+                const [foundedAccessRightSet] = await B2BAppAccessRightSet.getAll(support, {
+                    id: accessRightSet.id,
+                })
+                expect(foundedAccessRightSet).toHaveProperty('id', accessRightSet.id)
+            })
+
+            test('Can update', async () => {
+                const [updatedAccessRightSet] = await updateTestB2BAppAccessRightSet(support, accessRightSet.id)
+                expect(updatedAccessRightSet).toHaveProperty('id', accessRightSet.id)
+            })
+
+            test('Can soft-delete', async () => {
+                const [app] = await createTestB2BApp(admin)
+                const [createdAccessRightSet] = await createTestB2BAppAccessRightSet(admin, app)
+
+                const [deletedAccessRightSet] = await B2BAppAccessRightSet.softDelete(support, createdAccessRightSet.id)
+                expect(!!deletedAccessRightSet.deletedAt).toBeTruthy()
+            })
+
+            test('Can not hard-delete', async () => {
+                const [app] = await createTestB2BApp(admin)
+                const [createdAccessRightSet] = await createTestB2BAppAccessRightSet(admin, app)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await B2BAppAccessRightSet.delete(support, createdAccessRightSet.id)
+                })
+            })
+        })
+
+        describe('Service user', () => {
+            test('Can not create', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestB2BAppAccessRightSet(integratedServiceUser, app)
+                })
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestB2BAppAccessRightSet(integratedToAnotherAppServiceUser, app)
+                })
+            })
+
+            test('Can only read if integrated with app', async () => {
```

Comment:
What do you mean by term "integrated". App must be connected or it should have access rights?